internal/resources/aws/dynamodb_table.go

```diff
@@ -0,0 +1,336 @@
+package aws
+
+import (
+	"fmt"
+
+	"github.com/infracost/infracost/internal/schema"
+	"github.com/shopspring/decimal"
+)
+
+type DynamoDbTableArguments struct {
+	Address        string   `json:"address,omitempty"`
+	Region         string   `json:"region,omitempty"`
+	BillingMode    string   `json:"billingMode,omitempty"`
+	WriteCapacity  int64    `json:"writeCapacity,omitempty"`
+	ReadCapacity   int64    `json:"readCapacity,omitempty"`
+	ReplicaRegions []string `json:"replicaRegions,omitempty"`
+}
+
+func NewDynamoDBTable(args *DynamoDbTableArguments, u *schema.UsageData) *schema.Resource {
```

Comment:
@tim775 Just leaving my comment here too. I believe the `Resource Abstraction` should not care about what fields come from what file. So all it takes is a bunch of mandatory and optional fields. The caller of this function then can decide how to construct it.

If I want to make an example is let's say we have the resource `X` with fields of `A`, `B` and `C`. In terraform, fields `A` and `B` are present in the state file and field `C` comes from usage data. But, there's no guarantee that all providers are like terraform. Another provider like Pulumi might have support for field `C` in its own config and state and here's where things get super complicated if we take two different parameters.